### PR TITLE
Easiest usage of "embed"

### PR DESCRIPTION
**Pull request by neni**
Add new function ebitenutil.NewImageFromEmbedFile(embedFs embed.FS, path string) (*ebiten.Image, image.Image, error)

Change ebitenutil.NewImageFromURL(url string) (*ebiten.Image, error) to ebitenutil.NewImageFromURL(url string) (*ebiten.Image, image.Image, error)

Add "embed" example

Add "url" example

### CODE REVIEW

File: ebitenutil/loadembedimage.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	"image"
+
+	"github.com/hajimehoshi/ebiten/v2"
+)
+
+// NewImageFromEmbedFile loads the file from embed.FS with path and returns ebiten.Image and image.Image.
+//
+// Image decoders must be imported when using NewImageFromEmbedFile. For example,
+// if you want to load a PNG image, you'd need to add `_ "image/png"` to the import section.
+func NewImageFromEmbedFile(embedFs embed.FS, path string) (*ebiten.Image, image.Image, error) {
```
Comment (eliasdaler) [inline ebitenutil/loadembedimage.go:31]: This should probably be a `fs.FS`. No need to restrict to `embed.FS`
I'd also name this `NewImageFromFileSystem`.

File: examples/embed/main.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	"embed"
+	_ "image/jpeg"
+	"log"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/ebitenutil"
+)
+
+// Embedding files from assets directory
+//go:embed assets/*
```
Comment (eliasdaler) [inline examples/embed/main.go:30]: I feel like it'd be better to embed a single file here

File: examples/url/main.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
+// Copyright 2014 Hajime Hoshi
```
Comment (eliasdaler) [inline examples/url/main.go:1]: Not sure if this is needed. Plus, it depends on external resources...

**Review by eliasdaler — CHANGES_REQUESTED** [review]

File: ebitenutil/loadembedimage.go (hunk 1 above)
Comment (neni) [reply ebitenutil/loadembedimage.go:31]: That's absolutely true.

File: examples/embed/main.go (hunk 2 above)
Comment (neni) [reply examples/embed/main.go:30]: That's sure for a single file, but it is for show how to use embed.FS

File: examples/url/main.go (hunk 3 above)
Comment (neni) [reply examples/url/main.go:1]: The changes are cosmetic differences from original files.

**Review by neni — COMMENTED** [review]
Use fs.FS in place of embed.FS 
Rename NewImageEmbedFile() to NewImageFromFileSystem()

File: examples/embed/main.go (hunk 2 above)
Comment (eliasdaler) [reply examples/embed/main.go:30]: You can specify only one file and still use embed.FS
See the example here: https://pkg.go.dev/embed
```go
import "embed"

//go:embed hello.txt
var f embed.FS
data, _ := f.ReadFile("hello.txt")
print(string(data))
```

File: examples/url/main.go (hunk 3 above)
Comment (eliasdaler) [reply examples/url/main.go:1]: These are not changes, these are additions. And this is a code duplication which doesn't act as a useful example, in my opinion.
I'll let Hajime decide if this is needed or not.

**Review by neni — COMMENTED** [review]
Fix "highdpi" example

File: examples/url/main.go (hunk 3 above)
Comment (neni) [reply examples/url/main.go:1]: I agree, "url" Example is not needed (I thought you were talking of "Copyright 2014 Hajime Hoshi").

**Comment by hajimehoshi** [comment]
* `go:embed` is not allowed yet. See https://github.com/hajimehoshi/ebiten/issues/1258
* Breaking changes in existing APIs are not allowed.
* In general, please create an issue first if you want to suggest a new API.

**Pull request closed without merging**